api.go

```diff
@@ -270,25 +270,39 @@ func Shun(fn any) Provider {
 // all values must be consumed.
 //
 // When used on an existing Provider, it creates an annotated copy of that provider.
-func MustConsume(fn any) Provider {
+//
+// MustConsume may be callled called on the provider it returns creating a provider
+// that has multiple MustConsume annotations.
+func MustConsume[T any](fn any) Provider {
```

Comment:
It's a shame with this signature that T cannot be inferred. It would be nice if your earlier example, `MustConsume[string](func() string)` could be just `MustConsume(func() string)`.

Reply:
It is too bad.